File: Retos/Reto #32 - LA COLUMNA DE EXCEL [Media]/python/EspinoLeandroo.py
def excel_column(input: str):
    column = 0;
    
    if(input):
        input = input.upper()
        
        i = 0
        for letra in input:
            column += (ord(letra) - 64) * (26 ** (len(input)-1-i))
            i += 1
            
        print(input + " => " + str(column))
        
        
excel_column("A")
excel_column("Z")
excel_column("AA")
excel_column("CA")
excel_column("ZYX")
excel_column("AYK")
excel_column("")

File: Retos/Reto #32 - LA COLUMNA DE EXCEL [Media]/java/EspinoLeandroo.java
public class EspinoLeandroo {

    public static void main(String[] args) {
        EspinoLeandroo espinoLeandroo = new EspinoLeandroo();

        espinoLeandroo.excelColumn("A");
        espinoLeandroo.excelColumn("Z");
        espinoLeandroo.excelColumn("AA");
        espinoLeandroo.excelColumn("CA");
        espinoLeandroo.excelColumn("ZYX");
        espinoLeandroo.excelColumn("AYK");
        espinoLeandroo.excelColumn("");

    }

    public void excelColumn(String input){
        int column = 0;

        if(!input.isBlank()){
            input = input.toUpperCase();
            for(int i = 0; i < input.length(); i++){
                column += (input.charAt(i) - 64) * (Math.pow(26, (input.length()-1-i)));
            }        
            System.out.println(input + " => " + column);
        }        
    }
}

